libcontainer/cgroups/fs/hugetlb.go

```diff
@@ -27,11 +27,30 @@ func (s *HugetlbGroup) Apply(d *cgroupData) error {
 	return nil
 }
 
+// HasReservationAccountingSupport checks if reservation accounting of huge pages in the hugetlb cgroup
+// is supported. This is supported from linux 5.7
+// https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v1/hugetlb.html
+func (s *HugetlbGroup) HasReservationAccountingSupport(path string) bool {
+	if len(HugePageSizes) == 0 {
+		return false
+	}
+	_, err := fscommon.ReadFile(path, strings.Join([]string{"hugetlb", HugePageSizes[0], "rsvd", "limit_in_bytes"}, "."))
+	return err == nil
+}
+
 func (s *HugetlbGroup) Set(path string, cgroup *configs.Cgroup) error {
+	supportsReservationAccounting := s.HasReservationAccountingSupport(path)
```

Comment:
Not sure if this is the best way to check, or should we try to "cache" the value like we do with `HugePageSizes`?